datafusion/expr/src/built_in_function.rs

```diff
@@ -895,15 +894,37 @@ impl BuiltinScalarFunction {
                 ],
                 self.volatility(),
             ),
-            BuiltinScalarFunction::DateTrunc => Signature::one_of(
-                vec![
-                    Exact(vec![Utf8, Timestamp(Nanosecond, None)]),
-                    Exact(vec![Utf8, Timestamp(Microsecond, None)]),
-                    Exact(vec![Utf8, Timestamp(Millisecond, None)]),
-                    Exact(vec![Utf8, Timestamp(Second, None)]),
-                ],
-                self.volatility(),
-            ),
+            BuiltinScalarFunction::DateTrunc => {
+                let time_zones = vec![
```

Comment:
Maybe @waitingkuo  has some ideas about how to deal with timezones in function signatures. 

It seems not quite right to have to list all timezones specially for `date_trun` only -- shouldn't the timezone logic apply to any function that takes timezones ? 🤔

Reply:
I am not sure there is a better way to provide the timezone here. And wait for @waitingkuo suggestion.

Also, it should be extended to other functions using timezone.

Reply:
Hi @alamb  @Weijun-H @tustvold . i don't have the solution. actually i have some other pr blocked by this for now. 
i wonder whether force it to be coerced to  UTC ```Timestamp(someunit, Some("+00:00".into())``` makes sense before we have better solution for what @tustvold  suggested